test/internet/test-tls-socket-constructor-alpn-npn-options-parsing.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+// Test that TLSSocket can take
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const tls = require('tls');
+
+new tls.TLSSocket(null, {
+  ALPNProtocols: ['http/1.1'],
+  NPNProtocols: ['http/1.1']
+});
```

Comment:
What I mean is that `options.ALPNProtocols` and `options.NPNProtocols` should still be arrays-of-strings afterwards.  Making  a copy with `Object.assign()` should do it.

Would it be possible to flesh out the test somewhat and verify that the options are present in the TLS handhake?  You'll need to guard on `process.features.tls_alpn` and `process.features.tls_npn`.

Reply:
I've got the skeleton of this, but I'm not sure where the TLS handshake itself happens (the docs have me tied in knots a bit). I think I should be looking at the `secureConnection` event. If that's the case, I need to create a server via the `TLSSocket` constructor and `net.createServer`. Am I on the right track?

Reply:
Yep.  If you grep for TLSSocket in test/parallel, you should find some existing examples.

Reply:
Yup. That's where I am now (I'll move this test file to that directory too if you've no objections).